visualisation/visualise-submissions/visualise-submission-functions.R

```diff
@@ -0,0 +1,142 @@
+plot_forecasts = function(national = TRUE,
+                          cutoff = 0){
+  
+  # Get observed data ------------------------------------------------------------------
+  source(here::here("utils", "get-us-data.R"))
+  
+  daily_deaths_state <- get_us_deaths(data = "daily") %>%
+    dplyr::mutate(day = ordered(weekdays(as.Date(date)), 
+                                levels=c("Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday")),
+                  epiweek_day = as.numeric(paste0(epiweek, ".", as.numeric(day)))) %>%
+    # filter out data from the last incomplete week
+    filter(epiweek_day < max(epiweek))
+  
+  
+  weekly_deaths_state <- daily_deaths_state %>%
+    dplyr::mutate(epiweek = lubridate::epiweek(date)) %>%
+    dplyr::group_by(state, epiweek) %>%
+    dplyr::summarise(deaths = sum(deaths),
+                     target_end_date = max(date)) %>%
+    dplyr::ungroup()
+  
+  weekly_deaths_national <- weekly_deaths_state %>%
+    dplyr::group_by(epiweek, target_end_date) %>%
+    dplyr::summarise(deaths = sum(deaths)) %>%
+    dplyr::ungroup()
+  
+  
+  # Get forecasts -----------------------------------------------------------
+  
+  ## Get most recent Rt forecast 
+  rt_forecasts <- readr::read_csv(here::here("rt-forecast", "submission-files",
+                                             "latest-rt-forecast-submission.csv")) %>%
+    dplyr::mutate(model = "Rt")
+  
+  ## Get timeseries forecasts
+  ts_deaths_only <- readr::read_csv(here::here("timeseries-forecast", "deaths-only",
+                                               "submission-files",
+                                               "latest-weekly-deaths-only.csv")) %>%
+    dplyr::mutate(model = "TS deaths")
+  
+  ts_deaths_on_cases <- readr::read_csv(here::here("timeseries-forecast", "deaths-on-cases",
+                                                   "submission-files", 
+                                                   "latest-weekly-deaths-on-cases.csv")) %>%
+    dplyr::mutate(model = "TS deaths on cases")
+  
+  ## Get mean average ensemble
+  mean_ensemble <- readr::read_csv(here::here("ensembling", "quantile-average",
+                                              "submission-files",
+                                              "latest-epiforecasts-ensemble1-qa.csv")) %>%
+    dplyr::mutate(model = "Mean ensemble")
+  
+  ## Get QRA ensemble
+  # qra_ensemble <- readr::read_csv(here::here("ensembling", "qra-ensemble",
+  #                                        "submission-files",
+  #                                        "latest-epiforecasts-ensemble1-qra.csv")) %>%
+  #   dplyr::mutate(model = "QRA ensemble")  
+  
+  
+  # Join forecasts ----------------------------------------------------------
+  # and add state names
+  forecasts <- dplyr::bind_rows(rt_forecasts, ts_deaths_only, ts_deaths_on_cases, #qra_ensemble
+                                mean_ensemble) %>%
+    dplyr::left_join(tigris::fips_codes %>%
```

Comment:
this fips thing happens again and again across the code base. Could do with being a utility function